support/sier-codec/src/lib.rs

```diff
@@ -48,13 +52,17 @@ impl Parser {
     pub fn struct_def(&self, name: &str) -> Option<&Arc<StructDef>> {
         self.structs.values().find(|s| s.type_name() == name)
     }
-}
 
-impl Default for Parser {
-    fn default() -> Self {
-        Parser {
-            structs: HashMap::new(),
-        }
+    pub fn parse_json<'a>(
+        &'a self,
+        file_json_contents: &str,
+        def: &'a Arc<StructDef>,
+    ) -> Result<Object, Error<'a>> {
+        let json: SerdeValue =
+            serde_json::from_str(file_json_contents).or(Err(Error::InvalidJson))?;
```

Comment:
`or(Err(...))` should be `map_err` I think. I'm surprised clippy doesn't say something about it.

Reply:
Actually, I don't know if clippy runs against this directory...

Reply:
Clippy runs and isn't finding any problems. Isn't map_err accepting a Function? Here I just need to transform the error, I am not sure which is the best way?